buildroot-external/package/lxd-guest-agent/99-lxd-agent.rules

```diff
@@ -0,0 +1 @@
+ACTION=="add", SYMLINK=="virtio-ports/org.linuxcontainers.lxd", TAG+="systemd", ACTION=="add", RUN+="/bin/systemctl start lxd-agent.service"
```

Comment:
Can we not use systemd unit VM detection like all other agent using?

Reply:
What would you recommend? A separate check for "/dev/lxd/sock" (indicating we are in a guest LXD) that is called by systemd?

Reply:
No, systemd can have conditions in descriptive form. There is `ConditionVirtualization=` which can be used to detect virtualization. So we can just set that in `lxd-agent.service`, I think, but we should test it with and without LXC if things behave as expected.

https://www.freedesktop.org/software/systemd/man/systemd.unit.html#ConditionVirtualization=
https://www.freedesktop.org/software/systemd/man/systemd-detect-virt.html#

Reply:
I understand now. Hmm, looking my test system there is not a good way to detect a LXD VM specifically, systemd-detect-virt shows just qemu. Apologies, I scraped this together from the debian package lxd-agent-loader. Will dig in to see if I can better detect this. Would it not be a good enough test to check for /dev/lxd/sock in this case if the existing systemd detection is not good enough?

Reply:
Also, we may consider updating the virtualization checks for the qemu agent, as that tries and fails to load in this setup.

Reply:
> systemd-detect-virt shows just qemu

Hm, I see there is no `lxd` :cry: 

Ideally we'd add lxd to systemd-detect-virt, then the Qemu problem would be solved as well :smile:

Reply:
I just looked at the systemd source it looks like they are detecting /sys/class/dmi/id/sys_vendor which shows as. I just looked and LXD shows up properly in /sys/class/dmi/id/board_name. 

I can try and create a PR for systemd, but that is a tall order. Any chance we can implement a hack and I can follow up later with the correct systemd PR?

Reply:
I'd prefer if we have a PR with systemd already open before patching it on our side, so the process is at least kicked off :)

Seems that worth a try submitting a PR which just reads `/sys/class/dmi/id/board_name`?

Reply:
We can apply the PR also into our build system, so we not have to wait to get approved by systemd @bryanyork

Reply:
Working systemd patchset added. We are behind the latest systemd version and they refactored things. I'll work on a PR to systemd for HEAD now.